addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -4262,6 +4268,31 @@ QUnit.test('mark channel as seen if last message is visible when switching chann
     );
 });
 
+QUnit.test('add custom filter should filter messages accordingly to selected filter', async function (assert) {
+    assert.expect(4);
+
+    this.data['mail.channel'].records.push({
+        id: 20,
+        name: "General"
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_fetch') {
+                const domainsAsStr = args.kwargs.domain.map(domain => domain.join(''));
+                assert.step(`message_fetch_${domainsAsStr.join('')}`);
+            }
+            return this._super(...arguments);
+        },
+    });
+    assert.verifySteps(['message_fetch_needaction=true'], "A message_fetch request should have been done for needaction messages");
+
+    // Open filter menu of control panel and select a custom filter (id = 0)
+    await toggleFilterMenu(document.body);
+    await toggleAddCustomFilter(document.body);
+    await applyFilter(document.body);
```

Comment:
not your fault but those are very unclear and the api is strange too (`document.body`?!)

Reply:
It needs an element to work, so I gave body ..

Reply:
Ok, I think we are meant to send `this.widget` just in case there are somehow multiple control panels on the page, but I guess `body` is fine too since at the end we always use `body` in our new tests so that's consistent with what we do.